lighthouse-core/audits/byte-efficiency/unused-css-rules.js

```diff
@@ -66,14 +61,41 @@ class UnusedCSSRules extends ByteEfficiencyAudit {
       }
 
       if (rule.used) {
-        stylesheetInfo.used.push(rule);
-      } else {
-        unused++;
-        stylesheetInfo.unused.push(rule);
+        stylesheetInfo.usedRules.push(rule);
       }
     });
+  }
+
+  /**
+   * @param {!Object} stylesheetInfo
+   * @return {{wastedBytes: number, totalBytes: number, wastedPercent: number}}
+   */
+  static computeUsage(stylesheetInfo) {
+    let usedUncompressedBytes = 0;
+    const totalUncompressedBytes = stylesheetInfo.content.length;
 
-    return unused;
+    for (const usedRule of stylesheetInfo.usedRules) {
+      usedUncompressedBytes += usedRule.endOffset - usedRule.startOffset;
+    }
+
+    const networkRecord = stylesheetInfo.networkRecord;
+    let totalBytes = networkRecord && stylesheetInfo.networkRecord.transferSize;
+    if (!networkRecord || networkRecord._resourceType._name !== 'stylesheet') {
+      // If we don't know for sure how many bytes this sheet used on the network,
+      // we can guess it was roughly the size of the content length.
+      const wasCompressed = !networkRecord ||
+          networkRecord._transferSize !== networkRecord._resourceSize;
+      totalBytes = wasCompressed ? Math.round(totalUncompressedBytes / 3) : totalUncompressedBytes;
+    }
```

Comment:
this is a little confusing to follow. What do you think of something like `totalTransferredBytes` for `totalBytes` and then maybe folding the first short circuit into the conditional like
```js
let totalTransferredBytes;
if (networkRecord && networkRecord._resourceType._name === 'stylesheet') {
  totalTransferredBytes = networkRecord.transferSize;
} else {
  // If we don't know for sure how many bytes this sheet used on the network,
  // we can guess it was roughly the size of the content length.
  const wasCompressed = !networkRecord ||
      networkRecord._transferSize !== networkRecord._resourceSize;
  totalTransferredBytes = wasCompressed ? Math.round(totalUncompressedBytes / 3) :
      totalUncompressedBytes;
}
```
the variable rename alone might be enough to clear things up, though